src/map.c

```diff
@@ -99,30 +99,24 @@ SEXP map_impl(SEXP env, SEXP x_name_, SEXP f_name_, SEXP type_, SEXP progress) {
   return out;
 }
 
-SEXP map2_impl(SEXP env, SEXP x_name_, SEXP y_name_, SEXP f_name_, SEXP type_, SEXP progress) {
-  const char* x_name = CHAR(Rf_asChar(x_name_));
-  const char* y_name = CHAR(Rf_asChar(y_name_));
-  const char* f_name = CHAR(Rf_asChar(f_name_));
-
-  SEXP x = Rf_install(x_name);
-  SEXP y = Rf_install(y_name);
-  SEXP f = Rf_install(f_name);
+SEXP map2_impl(SEXP env, SEXP type_, SEXP progress, SEXP error_call) {
+  SEXP x = Rf_install(".x");
+  SEXP y = Rf_install(".y");
+  SEXP f = Rf_install(".f");
   SEXP i = Rf_install("i");
   SEXPTYPE type = Rf_str2type(CHAR(Rf_asChar(type_)));
 
   SEXP x_val = PROTECT(Rf_eval(x, env));
-  check_vector(x_val, ".x", env);
+  check_vector(x_val, ".x", error_call);
   SEXP y_val = PROTECT(Rf_eval(y, env));
-  check_vector(y_val, ".y", env);
+  check_vector(y_val, ".y", error_call);
 
   int nx = Rf_length(x_val), ny = Rf_length(y_val);
   if (nx != ny && nx != 1 && ny != 1) {
-    r_abort(
-      "Mapped vectors must have consistent lengths:\n"
-      "* `.x` has length %d\n"
-      "* `.y` has length %d",
-      nx,
-      ny
+    r_abort_call(
+      error_call,
+      "`.y must have length 1 or %i, not %i.",
```

Comment:
```suggestion
      "`.y` must be length 1 or %i, not %i.",
```

More consistent with `Result must be length 1` error from above

Reply:
(at the very least, catch the missing backtick on `.y`)

Reply:
I will fix the backtick though.